src/SubspaceClustering.jl

```diff
@@ -2,4 +2,75 @@ module SubspaceClustering
 
 # Write your package code here.
 
+## K-Subspaces SubspaceClustering
+
+#Imports
+import LinearAlgebra: norm, svd, transpose
+import Base: copy, deepcopy, findall, view
+import Base:argmax
+using Random: randn
+using Logging: @info
+using ProgressLogging: @progress
+using ArnoldiMethod: partialschur
+
+#Exports
+export polar, KSS
+
+#Polar function
+
+function polar(X)
+	U, _, V = svd(X)
+	U*V'
+end
+
+"""
+	KSS(X, d; niters=100)
+
+Run K-subspaces on the data matrix `X`
+with subspace dimensions `d[1], ..., d[K]`,
+treating the columns of `X` as the datapoints.
+"""
+function KSS(X, d; niters=100, Uinit=polar.(randn.(size(X, 1), collect(d))))
+	K = length(d)
+	D, N = size(X)
+
+	# Initialize
+	U = deepcopy(Uinit)
+	c = [argmax(norm(U[k]' * view(X, :, i)) for k in 1:K) for i in 1:N]
+	c_prev = copy(c)
+
+	# Iterations
+	@progress for t in 1:niters
+		# Update subspaces
+		for k in 1:K
+			ilist = findall(==(k), c)
+			# println("Cluster $k, ilist size: ", length(ilist))
```

Comment:
It would be great to handle things like this via the built-in logging functionality: https://docs.julialang.org/en/v1/stdlib/Logging/

See here also: https://modernjuliaworkflows.org/writing/#logging

Reply:
This is still commented out. Let's make this a `@debug`